src/OpenLoco/src/Input/MouseInput.cpp

```diff
@@ -2292,4 +2299,70 @@ namespace OpenLoco::Input
             return MouseButton::released;
         }
     }
+
+    // 0x004C6202
+    void processMouseWheel()
+    {
+        int wheel = 0;
+
+        while (true)
+        {
+            _cursorWheel -= 1;
+
+            if (_cursorWheel < 0)
+            {
+                _cursorWheel = 0;
+                break;
+            }
+
+            wheel -= 17;
+        }
+
+        while (true)
+        {
+            _cursorWheel += 1;
+
+            if (_cursorWheel > 0)
+            {
+                _cursorWheel = 0;
+                break;
+            }
+
+            wheel += 17;
+        }
+
+        if (Tutorial::state() != Tutorial::State::none)
+            return;
+
+        if (Input::hasFlag(Input::Flags::rightMousePressed))
+        {
+            if (OpenLoco::isTitleMode())
+                return;
+
+            auto main = WindowManager::getMainWindow();
+            if (main != nullptr && wheel != 0)
+            {
+                if (wheel > 0)
+                {
+                    main->viewportRotateRight();
+                }
+                else if (wheel < 0)
+                {
+                    main->viewportRotateLeft();
+                }
+                TownManager::updateLabels();
+                StationManager::updateLabels();
+                Windows::MapWindow::centerOnViewPoint();
+            }
```

Comment:
Could this be better delegated to the window manager as well?

Reply:
Most likely, not for this PR, been meaning to create a chart first to see where is what and this PR is rather a side effect of me wanting to get the mouse wheel working correctly with the bonus of cleanup.